tests/Model/SearchIndexInputTest.php

```diff
@@ -32,6 +32,14 @@ public function testConstructorShouldRequireNameToBeString($name): void
         new SearchIndexInput(['definition' => ['mapping' => ['dynamic' => true]], 'name' => $name]);
     }
 
+    /** @dataProvider provideInvalidStringValues */
+    public function testConstructorShouldRequireTypeToBeString($type): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Expected "type" option to have type "string"');
+        new SearchIndexInput(['definition' => ['mapping' => ['dynamic' => true]], 'type' => $type]);
+    }
```

Comment:
@GromNaN: Added a unit test here and in `CreateSearchIndexesTest` below. The test in the other file seems redundant, but we already had a test for `name` so I thought it best to preserve the consistency.